packages/keystone/src/lib/core/where-inputs.ts

```diff
@@ -62,8 +62,10 @@ export async function resolveWhereInput(
           };
         }
         const field = list.fields[fieldKey];
-        // we know if there are filters in the input object with the key of a field, the field must have defined a where input so this non null assertion is okay
-        const where = field.input!.where!;
+        const where = field.input!.where;
+        if (!where) {
+          throw new Error(`The field "${list.listKey}.${fieldKey}" does not support filtering.`);
+        }
```

Comment:
When will this happen?

Reply:
This happens when someone is using the items API and they use a filter which doesn't exist.

Reply:
But if the filter doesn't exist, they'll get a validation error from GraphQL and never get there?

Reply:
Let me pull up an example, I was able to trigger this exact code path while running tests that hadn't been updated to enable filters.

Reply:
Ah yeah, it's in access control, if you return declarative access control (e.g. a filter) it will send the returned object through this code path, at which point we catch the error.

Reply:
We run GraphQL validation on filters returned from access control though so this still shouldn't be hit, do you have a test I could have a look at?